packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -191,94 +183,63 @@ const messageContentAriaText = (props: ChatMessageContentProps): string | undefi
-      return node.name === 'msft-mention';
-    }
-    return false;
-  },
-  processNode: (node) => {
-    if (props.mentionDisplayOptions?.onRenderMention) {
-      const { id } = node.attribs;
-      const mention: Mention = {
-        id: id,
-        displayText: node.children[0]?.data ?? ''
-      };
-      return props.mentionDisplayOptions.onRenderMention(mention, defaultOnMentionRender);
-    }
-    return processNodeDefinitions.processDefaultNode;
-  }
-});
-
 const processHtmlToReact = (props: ChatMessageContentProps): JSX.Element => {
-  const steps: ProcessingInstructionType[] = [
-    processInlineImage(props),
-    /* @conditional-compile-remove(mention) */
-    processMention(props),
-    {
-      // Process everything else in the default way
-      shouldProcessNode: IsValidNodeDefinitions.alwaysValid,
-      processNode: processNodeDefinitions.processDefaultNode
+  const options: HTMLReactParserOptions = {
+    transform(reactNode, domNode) {
+      if (domNode instanceof DOMElement && domNode.attribs) {
+        // Transform custom rendering of mentions
+        /* @conditional-compile-remove(mention) */
+        if (props.mentionDisplayOptions?.onRenderMention && domNode.name === 'msft-mention') {
+          const { id } = domNode.attribs;
+          const mention: Mention = {
+            id: id,
+            displayText: (domNode.children[0] as unknown as Text).nodeValue ?? ''
+          };
+          return props.mentionDisplayOptions.onRenderMention(mention, defaultOnMentionRender);
+        }
+
+        // Transform inline images
+        if (
+          domNode.name &&
+          domNode.name === 'img' &&
+          domNode.attribs &&
+          domNode.attribs.id &&
+          props.message.inlineImages?.find((metadata) => {
+            return metadata.id === domNode.attribs.id;
+          })
+        ) {
+          domNode.attribs['aria-label'] = domNode.attribs.name;
+          // logic to check id in map/list
+          if (props.attachmentsMap && domNode.attribs.id in props.attachmentsMap) {
+            domNode.attribs.src = props.attachmentsMap[domNode.attribs.id];
+          }
+          const handleOnClick = (): void => {
+            props.onInlineImageClicked && props.onInlineImageClicked(domNode.attribs.id);
+          };
+          const imgProps = attributesToProps(domNode.attribs);
+          return (
+            <span
+              data-ui-id={domNode.attribs.id}
+              onClick={handleOnClick}
+              tabIndex={0}
+              role="button"
+              style={{
+                cursor: 'pointer'
+              }}
+              onKeyDown={(e) => {
+                if (e.key === 'Enter') {
+                  handleOnClick();
+                }
+              }}
+            >
+              <img {...imgProps} />
+            </span>
+          );
+        }
+      }
+      // Pass through the original node
```

Comment:
The casting is needed as the underlying library wraps this in a function to hide the casting to `any`.

https://github.com/remarkablemark/html-react-parser/blob/master/src/dom-to-react.ts#L36C35-L36C35